src/pages/auth/Login.jsx

```diff
@@ -0,0 +1,71 @@
+import { Link } from "react-router-dom"
+import { useState } from "react"
+import "./auth.css"
+import { useAuth } from "../../context/auth-context"
+export const Login = () => {
+
+    const { userlogin } = useAuth()
+    const [loginDetail, setloginDetail] = useState({
+        email: "",
+        password: "",
+    })
+
+    return (
+        <div>
+            <main>
+                <div className="modal-container  auth-position">
+                    <form action="" className="auth-wrapper" onSubmit={(e) => {
+                        e.preventDefault()
+                        userlogin(loginDetail)
+                    }}>
+                        <div>
+                            <h2 className="text-center">login</h2>
+                        </div>
+                        <div className="flex-col">
+                            <label htmlFor="email">Email address</label>
+                            <input type="email" id="email" className="fs-sm input-padding" autoComplete="off" required
+                                onChange={(e) => {
+                                    setloginDetail((prev) => {
+                                        return {
+                                            ...prev,
+                                            email: e.target.value
+                                        }
+                                    })
+                                }}
+                            />
+                        </div>
+                        <div className="flex-col">
+                            <label htmlFor="password">Password</label>
+                            <input type="password" id="password" className="fs-sm input-padding" required autoComplete="off"
+                                onChange={(e) => {
+                                    setloginDetail((prev) => {
+                                        return {
+                                            ...prev,
+                                            password: e.target.value
+                                        }
```

Comment:
Suggestion:
the function in the onChange can be abstracted as a `signupChangeHandler` as following
`const signupChangeHandler = (e) => {
    setLoginDetails((prev) => {
        return {...prev,[e.target.name]:e.target.value}
    })
}` so that you can avoid repeating  yourself

Reply:
yes that's a good suggestion
thanks Nikhil